addons/mail/static/src/model/model_manager.js

```diff
@@ -892,6 +919,64 @@ export class ModelManager {
         return inverseField;
     }
 
+    /**
+     * Marks a Model on which all() is being used.
+     *
+     * @private
+     * @param {Object} Model
+     */
+    _markAllBeingUsed(Model) {
+        for (const listener of this._listeners) {
+            this._listenersUsingModelAll.get(Model).add(listener);
+        }
+    }
+
+    /**
+     * Marks a record as being used.
+     *
+     * @private
+     * @param {ModelField} field
+     * @param {string} localId
+     */
+    _markFieldBeingUsedOnLocalId(field, localId) {
+        for (const listener of this._listeners) {
+            this._localIdsUsedByListener.get(listener).add(localId);
+            this._listenersUsingLocalId.get(localId).add(listener);
+            this._listenersUsingLocalIdField.get(localId).get(field).add(listener);
+        }
+    }
+
+    /**
+     * Marks the given listener to be notified.
+     *
+     * @private
+     * @param {Object} listener
+     */
+    _markListenerToNotify(listener) {
+        if (listener.isPartOfUpdateCycle) {
+            this._listenersToNotifyInUpdateCycle.add(listener);
+        }
+        if (!listener.isPartOfUpdateCycle) {
+            this._listenersToNotifyAfterUpdateCycle.add(listener);
+        }
+    }
+
+    /**
+     * Marks a record as being used.
+     *
+     * @private
```

Comment:
As a general remark, the docstring are not good. They are basically the human format of the method name, without giving proper documentation to what this is (and why/extra info when things are sophisticated by design, which is very much the case here)

Reply:
Those methods are making it more confusing than it should be with having indirection, I'll just remove most of them and put the code where it belongs.